### DISCUSSION TITLE
Duration struct

### DISCUSSION THREAD

**Discussion by younies** [unicode-org/icu4x#4955] [Duration Formatter]
Hi @sffc 

@DesmondWillowbrook and @younies think we should create a struct called `Duration` to set the duration values for the input to the `DurationFormatter`. Or there is another `Duration` object that we can use.

**Comment by sffc** [comment]
To get started, I think `duration` should define its own struct with the 9 fields required by Intl.DurationFormat, probably each of type `u64`, which is a bit big but it's fine for a short-lived object. Eventually we might want to add a trait or conversions into other Duration types. We already have 2 other duration types in ICU4X, one in the `ixdtf` crate and another in the `icu_calendar` crate, but they are both very specific to their use cases.

CC @Manishearth

**Reply by sffc** [reply]
CC @jedel1043
